jd-geek-source/src/app/components/common/product-display-portal/product-display-portal.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { CommonPortalData } from "src/app/models/commonPortalData.interface";
+import { BestBuyService } from "src/app/services/best-buy.service";
+
+const RIGHT_SCROLL_STOP = 0;
+
+@Component({
+  selector: "app-product-display-portal",
+  templateUrl: "./product-display-portal.component.html",
+  styleUrls: ["./product-display-portal.component.scss"],
+})
+export class ProductDisplayPortalComponent implements OnInit {
+  @Input() productData: CommonPortalData[];
+  @Input() carouselID: string;
+  carouselElementID: string;
+  portalProducts: CommonPortalData[] = null;
+  leftScrollDisabled: boolean;
+  rightScrollDisabled = true;
+  maxScroll: number;
+  scrollTracker = RIGHT_SCROLL_STOP;
+
+  constructor(private bestBuyService: BestBuyService) {}
+
+  ngOnInit(): void {
+    this.portalProducts = this.productData;
```

Comment:
there's no reason to create a separate field only to reassign the input.  just use productData.  this elimates the need for the code inside the onInit and onChanges entirely.

Reply:
I think I figured that out later...lol